libraries/AP_ToshibaCAN/AP_ToshibaCAN.cpp

```diff
@@ -532,7 +532,7 @@ void AP_ToshibaCAN::send_esc_telemetry_mavlink(uint8_t mav_chan)
                 voltage[j] = _telemetry[esc_id].voltage_cv;
                 current[j] = _telemetry[esc_id].current_ca;
                 current_tot[j] = constrain_float(_telemetry[esc_id].current_tot_mah, 0, UINT16_MAX);
-                rpm[j] = _telemetry[esc_id].rpm;
+                rpm[j] = MAX(_telemetry[esc_id].rpm, 0);    // mavlink message only accepts positive rpm values
```

Comment:
```suggestion
                rpm[j] = abs(_telemetry[esc_id].rpm, 0);    // mavlink message only accepts positive rpm values
```

This will allow us to extend this message to support reversing motors.  Those GCSs that don't understand the reverse mask will still see an RPM - just not know which way that is.  I think that's reasonable.

This branch contains suitable (untested but compiling) changes to add that reversed mask:

https://github.com/ArduPilot/ardupilot/compare/master...peterbarker:rmackay9-tcan-rpm-fix?expand=1